### PR TITLE
Added MyDNS.JP

### PR DESCRIPTION
**Pull request by epgdatacapbon**
Added support for https://www.mydns.jp

Closed the original pullrequest (https://github.com/ddclient/ddclient/pull/35),
and reopened to the latest commit.

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
You could have rebased https://github.com/ddclient/ddclient/pull/35 to master.

**Comment by epgdatacapbon** [comment]
Actually, I slightly modified the code.

**Comment by DaveSophoServices** [comment]
Thanks. Should be able to merge this in the near future. 

Dave

**Comment by sanpei3** [comment]
Unfortunately, v3.9.1 does not have this change. Internally I use this patched ddclient without an issue. I hope it will include the next release.

File: ddclient [hunk 1]
```diff
@@ -5264,6 +5277,77 @@ sub nic_dnsmadeeasy_update {
...
+  server=fqdn.of.service       ## defaults to www.mydns.jp.
+  login=service-login          ## login name and password registered with the service.
+  password=service-password    ##
+  fully.qualified.host         ## the host registered with the service.
+
+Example ${program}.conf file entries:
+  ## single host update
+  protocol=mydns,                                              \\
+  login=my-mydns.jp-login,                                     \\
+  password=my-mydns.jp-password                                \\
```
Comment (SuperSandro2000) [inline ddclient:5301]:
```suggestion
  protocol=mydns,                \\
  login=my-mydns.jp-login,       \\
  password=my-mydns.jp-password  \\
```

File: ddclient [hunk 2]
```diff
@@ -5264,6 +5277,77 @@ sub nic_dnsmadeeasy_update {
...
+
+        ## update the DNS record
+        my $url;
+        my $header;
+        my $data;
+        my $reply;
+        $url = "https://$config{$h}{'server'}/directip.html";
+        $header = 'Content-Type: application/x-www-form-urlencoded';
+        $data = "MID=" . $config{$h}{'login'} . "&PWD=" . $config{$h}{'password'} . "&IPV4ADDR=" . $ip;
+        $reply = geturl(opt('proxy'), $url, undef, undef, $header, 'POST', $data);
```
Comment (SuperSandro2000) [inline ddclient:5325]:
```suggestion
        my $url = "https://$config{$h}{'server'}/directip.html";
        my $header = 'Content-Type: application/x-www-form-urlencoded';
        my $data = "MID=" . $config{$h}{'login'} . "&PWD=" . $config{$h}{'password'} . "&IPV4ADDR=" . $ip;
        my $reply = geturl(opt('proxy'), $url, undef, undef, $header, 'POST', $data);
```

**Comment by SuperSandro2000** [comment]
Same as #92. If there are no objections I will merge this shortly.

**Comment by epgdatacapbon** [comment]
Thanks SuperSandro2000.
I have no objection.

File: ddclient [hunk 3]
```diff
@@ -493,6 +493,10 @@ my %variables = (
     'dondominio-common-defaults'       => {
         'server'	      => setv(T_FQDNP,  1, 0, 1, 'dondns.dondominio.com', undef),
     },
+    'mydns-common-defaults'   => {
+        'server'          => setv(T_FQDNP,  1, 0, 1, 'www.mydns.jp',      undef),
+        'max-interval'    => setv(T_DELAY,  0, 0, 1, interval('6d'),      0),
+    },
```
Comment (SuperSandro2000) [inline ddclient:499]: Can we move those variables into the service specific ones?

Comment (epgdatacapbon) [reply ddclient:499]: Yes. No problem.

**Comment by SuperSandro2000** [comment]
Closing due to inactivity. Feel free to comment and I can reopen the PR.

**Pull request closed without merging**